.github/actions/display-jest-test-summary/action.yml

```diff
@@ -0,0 +1,32 @@
+name: Display E2E Test Summary
+
+inputs:
+  test_report_path:
+    type: string
+    description: Path of test report JSON file
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Transform Input
+      id: transform_input
+      run: |
+        echo JSON_OUTPUT=$(  jq '{ numTotalTests, numPassedTests, numFailedTests}' ${{ inputs.test_report_path }} ) >> $GITHUB_OUTPUT
+      shell: bash
+    - name: Display Jest Test Summary
+      uses: actions/github-script@v6
+      with:
+        script: |
+          const input = ${{ steps.transform_input.outputs.JSON_OUTPUT }};          
```

Comment:
Same with this, store it in an env and reference it using `process.env`.

Reply:
Just this one that's left to do I think ^^ @alexisc-kainos

Reply:
@JoshuaLicense changed to write to GITHUB_ENV instead of output

Reply:
Apologies, I'm probably not explaining this very well. What we're trying to get rid of is `${{ ... }}` this part from any part of the action that's expanded and executed such as `run` or the `script` section. 

Using the `${{ }}` syntax in the `env` section is safer because the user-controlled string is stored in a variable instead.

To avoid this we can do this: https://github.com/actions/github-script?tab=readme-ov-file#use-env-as-input

So something like this:
```yaml
runs:
  using: 'composite'
  steps:
    - name: Transform Input
      id: transform_input
      env:
        INPUT_PATH: ${{ inputs.test_report_path }}
      run: |
        echo JSON_OUTPUT=$(  jq '{ numTotalTests, numPassedTests, numFailedTests}' $INPUT_PATH ) >> $GITHUB_OUTPUT
      shell: bash
    - name: Display Jest Test Summary
      uses: actions/github-script@v6
      env:
        JSON: ${{ steps.transform_input.outputs.JSON_OUTPUT }}
      with:
        script: |
          const input = process.env.JSON;

          await core.summary
              .addHeading(`Results for End to End Tests`)
              .addTable([
                [
                    {data: 'Total Tests', header: true}, 
                    {data: 'Tests Passed ✅', header: true}, 
                    {data: 'Tests Failed ❌', header: true},
                ],
                [ input.numTotalTests.toString(), input.numPassedTests.toString(), input.numFailedTests.toString(), ]
              ])
              .write()
```

Reply:
Apologies Josh, have updated the action. I think I used the {{ }} syntax so that I could keep the JSON as an object straight from step transform_input. But I can see how using the process.env is safer against injection. Thanks for the explanation 
Is this implementation okay now?

Reply:
Looks good! Thanks for that!

Reply:
Sorry, the process.env is referencing the wrong variable `JSON_OUTPUT` instead of `JSON` - Though I'm tired so correct me if I'm wrong 😄

Reply:
The process.env in the second step references the JSON variable which is in the env for that step, which references that JSON_OUTPUT which is the output variable from the first step (transform_input)

Reply:
You're right, but if we're sending to GitHub env in the first step we don't need the `env` on the next step setting a `JSON` environment variable.